dlib/CMakeLists.txt

```diff
@@ -34,18 +34,24 @@ endif()
 include(add_global_compiler_switch.cmake)
 
 
-# Make sure ENABLE_ASSERTS is defined for debug builds
-if (NOT CMAKE_CXX_FLAGS_DEBUG MATCHES "-DENABLE_ASSERTS")
-   set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DENABLE_ASSERTS" 
-      CACHE STRING "Flags used by the compiler during C++ debug builds." 
-      FORCE)
-endif ()
+if (DLIB_IN_PROJECT_BUILD)
```

Comment:
I'm not sure about details but just for your info you can add definition by [target_compile_definitions](https://cmake.org/cmake/help/v3.5/command/target_compile_definitions.html).